### DISCUSSION TITLE
Pyfluent Parametric study

### DISCUSSION THREAD

**Discussion by Paul090196** [ansys/pyfluent#1766] [Q&A] [answered]
Hello everyone,

I am currently working on automating the simulation process using PyFluent, but I've run into a few problems and could use some assistance. Specifically, I'm having trouble changing the residual value from 1e-3 to 1e-6 in my simulations. Additionally, I'm encountering issues while creating input and output parameters for a parametric study involving the analysis of drag and lift coefficients of an airfoil at three different angles.

To provide some context, I'm attempting to vary the angles of the airfoil and observe how it affects the drag and lift coefficients. In order to do this, I need to split the velocity magnitude into its x and y components, but I'm having difficulties creating these components as input parameters.

Furthermore, I'm facing a challenge when attempting to create an output parameter. Although I have successfully defined the report definition for the output parameter, it doesn't seem to work as expected.

If anyone has experience with PyFluent or similar tools and could offer some guidance, I would greatly appreciate it. Specifically, I would like to know how to change the residual value from 1e-3 to 1e-6, as well as how to properly create input parameters for the x and y components of the velocity magnitude. Additionally, any insights into creating effective output parameters for a parametric study would be invaluable.

Thank you in advance for your help and support!

Best regards,
Paul Stephen
![error2](https://github.com/ansys/pyfluent/assets/138394566/2dfd2008-5786-4542-b23e-c44e7ff16ba4)
![error1](https://github.com/ansys/pyfluent/assets/138394566/0f268792-9015-4103-9f28-d2eb078fe904)

**Comment by mkundu1** [comment]
For setting input and output parameters, you can follow [the example](https://parametric.fluent.docs.pyansys.com/version/stable/examples/00-parametric/parametric_static_mixer_1.html#create-input-parameters) from pyfluent-parametric.

**Reply by Paul090196** [reply]
I am writing to seek your assistance regarding two specific challenges I am facing while using PyFluent for my airfoil analysis.

Firstly, I am having difficulty assigning the components of velocity (x and y) as input parameters within my script. The documentation provides an example for assigning the magnitude of velocity as an input parameter, but in my case, I need to specify the components separately. Unfortunately, I couldn't find the appropriate keyword or syntax to accomplish this. Could you please guide me on how to assign the x and y components of velocity using the "Components" velocity specification method?

Secondly, I would like to change the default residual convergence value from 1e-3 to a more stringent value of 1e-6. However, I haven't been able to identify the method or parameter to modify this setting within the PyFluent script. I would greatly appreciate your guidance on how to change the residual convergence value to the desired 1e-6.

Your expertise and support in resolving these challenges would be immensely valuable to me. I am eager to continue my analysis and automation efforts, and your guidance will greatly facilitate my progress.

Thank you very much for your attention and support. I look forward to your response and guidance

**Reply by mkundu1** [reply]
The following script will set x-velocity to a parameter:
```
import ansys.fluent.core as pyfluent
from ansys.fluent.core import examples
import_filename = examples.download_file("Static_Mixer_main.cas.h5", "pyfluent/static_mixer")
solver = pyfluent.launch_fluent()
solver.file.read_case(file_name=import_filename)  
solver.tui.define.boundary_conditions.set.velocity_inlet("inlet1", (), "velocity-spec", "no", "yes", "quit")  # Set velocity-specification to components
solver.tui.define.boundary_conditions.set.velocity_inlet("inlet1", (), "direction-0", "yes", "vel-x", 10, "quit") # Set x-component of velocity to a parameter vel-x
```
The section [TUI Command Construction](https://fluent.docs.pyansys.com/version/stable/user_guide/tui_commands.html#tui-command-construction) describes how to construct PyFluent's TUI calls from Fluent's TUI calls. You can record the TUI calls in a journal while setting up part of the problem in Fluent TUI and then use the Fluent journal as a reference to deduce the TUI calls from PyFluent.

**Reply by Paul090196** [reply]
I am writing to express my gratitude for your valuable guidance in solving the issue related to assigning the components of velocity within my script. Your help has been instrumental in moving my project forward, and I truly appreciate your support.
However, I have encountered another challenge that I hope you can assist me with. I would like to modify the default residual convergence value from 1e-3 to a more stringent value of 1e-6. Unfortunately, I have been unable to identify the appropriate method or parameter to make this adjustment within the PyFluent script. Could you please provide me with further guidance on how to change the residual convergence value to the desired 1e-6? Any assistance or insights you can offer would be greatly appreciated.

Thank you once again for your exceptional support in resolving the first issue. I look forward to hearing from you soon and receiving your guidance regarding the second problem. Your expertise and assistance are truly invaluable.

**Reply by mkundu1** [reply, marked as answer]
`solver.tui.solve.monitors.residual.convergence_criteria(1e-6, 1e-6, 1e-6, 1e-6, 1e-6, 1e-6, 1e-6)` sets up the convergence criteria for a case I'm testing with. It is better to set this up in Fluent TUI first with turning on the journal recording and use the Fluent journal as reference to confirm the residual names and order.

**Reply by seanpearsonuk** [reply]
@h-krishnan @gyeole Do you know if `solver.tui.solve.monitors.residual.convergence_criteria` will be reflected in the settings API soon?

**Comment by Paul090196** [comment]
![parameter](https://github.com/ansys/pyfluent/assets/138394566/7ee499d1-edb0-47b9-b4e7-da268e828154)
Subject: Inquiry about Setting Rotational Velocity as an Input Parameter in
PyFluent

Dear Sir,

I hope this email finds you well. I am currently conducting a parametric
study using PyFluent for my simulation work. While working with the
software, I came across a query that I would greatly appreciate your
assistance with.

In the documentation for PyFluent, I noticed that velocity can be assigned
as an input parameter. However, my specific simulation requires the use of
a moving frame of reference, where rotational velocity is a crucial input
parameter. To achieve this, I would like to set the rotational velocity as
an input parameter in my simulations.

Interestingly, I was able to successfully make this adjustment using the
graphical user interface (GUI) provided by Fluent. However, I encountered
difficulties when attempting to achieve the same result through the
PyFluent scripting interface. Despite my efforts, I couldn't figure out the
proper method to set the rotational velocity as an input parameter using
the scripting approach.

I believe that having the ability to control this parameter
programmatically through the script is essential for automating my
parametric study effectively. Thus, I am reaching out to you for guidance
on how to achieve this using PyFluent's scripting capabilities.

To provide a clearer understanding of my query, I have attached an image
that visually represents the scenario I am describing.

I am confident that with your expertise, I will be able to overcome this
challenge and proceed with my simulation work as intended. Your assistance
would be of great value to me in advancing my research.

Thank you in advance for your time and support. I am looking forward to
your response.

Best regards,
Paul Stephen

**Reply by mkundu1** [reply]
Duplicate of https://github.com/ansys/pyfluent/discussions/1917
